android/app/src/main/kotlin/com/elodin/walkie_talkie/BluetoothLeAudioManager.kt

```
@@ -62,22 +65,35 @@ class BluetoothLeAudioManager(private val context: Context) {
     }
 
     init {
-        bluetoothAdapter?.getProfileProxy(context, profileListener, BluetoothProfile.LE_AUDIO)
+        // Only attempt LE Audio init if supported (API 33+, hardware-gated)
+        if (isLeAudioSupported()) {
+            try {
+                bluetoothAdapter?.getProfileProxy(context, profileListener, BluetoothProfile.LE_AUDIO)
+            } catch (e: SecurityException) {
+                Log.w(TAG, "BLUETOOTH_CONNECT permission denied during LE Audio init", e)
+                onError?.invoke("Bluetooth permission required for LE Audio")
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)